### ISSUE TITLE
The `/metadata` folder is missing in v6.0.0

### ISSUE THREAD

**Issue by YamiOdymel** [FortAwesome/Font-Awesome#18662]
**Describe the bug**
The `/metadata` folder that exists in v5.x but now it's gone.

**To Reproduce**
The v5.x `metadata` folder
https://github.com/FortAwesome/Font-Awesome/tree/7d3d774145ac38663f6d1effc6def0334b68ab7e/metadata

**Expected behavior**
The metadata contains the `.json`, `.yml` files for me to do some API intergrations and 
CI test works, it's kind of a dependency my program highly relies on. 

Hope it can be added back to the project, love the project and I'm using it everyday.

**Screenshots**
![image](https://user-images.githubusercontent.com/7308718/153128936-dd907137-9828-49b4-8ca3-648e3ff60446.png)

**Version and implementation**
Version: v6.0.0

**Related**
https://github.com/FortAwesome/Font-Awesome/issues/13116

**Comment by somethingSTRANGE** [comment]
My projects also relies on `icons.json`. Are there plans to add that to the v6 release?

**Comment by tagliala** [comment]
Hi,

metadata folder has been intentionally removed, but there are plans of bring it back

No ETA at the moment

**Comment by somethingSTRANGE** [comment]
Thanks for the update. I can stick with the 5.x Desktop release in the meantime.

In case it helps, I currently only use the `styles`, `unicode`, `label`, and `free` properties from each glyph, although `changes` and `search` seem like properties I could use in the future for inhouse apps/tools.

I have no need for the `ligatures`, `voted`, and `svg` properties.  _I would actually love a version that excluded the `svg` data, since that would probably drop the file size down less than 10% of its current size._

**UPDATE: Just wanted to mention that I primarily use Pro, but metadata for Free is desired as well. My tools support both releases (Pro and Free), and `icons.json` determines the glyphs available in the tools.**

**Comment by spmsh** [comment]
+1 my application heavily relies on metadata. 
I'm generating a multilingual iconpicker based on it.
Please add it back to PRO version as well. Thanks.

**Comment by geilt** [comment]
I also noticed this file was gone while upgrading to 6.0, and I heavily use it as well for allowing clients to select icons for different sections of our Platform. So now we have to make our own metadata?

**Comment by Molkobain** [comment]
+1 we are using it internally for an icon picker. Please consider putting it back.

**Comment by AgentSmith0** [comment]
Please bring it back as soon as possible...

**Comment by jwr** [comment]
Hi — I use FontAwesome in a ClojureScript application and wrote a converter that generates Clojure code from the `icons.json` data. This lets me use the icons (as macros) in ClojureScript code and the compiler optimizes the usage and only includes those icons that are used. No third-party JS, no additional loading, no additional requests, the SVG strings are included straight in the highly-optimized compiled JavaScript. This has worked very well for me over the last years.

Without the `icons.json` file, FontAwesome 6 is not useful to me at all, unfortunately. Please bring it back, my "pro" subscription doesn't make sense without it.

**Comment by meiyasan** [comment]
I heavily use the metadata.yaml to avoid extra connections, cache some icon information, and let user select icons through a form. Hope there will be a quick fix soon..

**Comment by blainehubbard** [comment]
+1 My company's application heavily relies on metadata.
We generate an iconpicker based on it.
Please add it back ASAP.

**Comment by DanielMoberly** [comment]
The entire Font Awesome drupal module relies on this feature to support the 66 thousand+ site installations

https://www.drupal.org/project/fontawesome/issues/3179775

**Comment by tagliala** [comment]
Hi again,

this issue has been discussed for a while in the Pro Repository at FortAwesome/Font-Awesome-Pro#1577.

Now that FA6 has been released and this issue also affects the free version, it is probably better to continue here the discussion, and I thought it was important to share this comment by @sensibleworld 

> Hey folks. Wanted to follow-up here with where we are on this decision.
> 
> The long and short of it is that we need to decouple our metadata from our release cycles. Our icon search is something that we’ve worked hard to improve for version 6, and we hope to continue to improve it as we move forward. Having it tied to releases just doesn’t make a ton of sense to us.
> 
> Case in point, we were literally in the process of starting the full build for our official v6.0 release this morning, and I noticed that our new `shrimp` icon was missing from our `maritime` category. In and of itself, this isn’t a huge deal, but it’s an omission I’d like to avoid. So, I had to quickly make the edit and get a new PR up and approved in order to make the release train. It’s a whole added layer of complexity that just doesn’t need to be there. We’ve also had a number of cases (and will likely continue to have cases) where we have the wrong search terms, or having missing search terms, in our data. We need to be able to make those changes without releasing a full version of Font Awesome. And we want any changes we make to be available to everyone as soon as we make them, which is why we want folks to leverage the new API.
> 
> We really do hope and believe that the GraphQL API is the best way to get up-to-date metadata about our icons (with a cached version as fallback). But we also recognize that static yaml and json files can be easier to use in some cases. In order to bridge the gap between the two, what we'd like to do is create a script for folks to generate those files directly from our API. You'll need to regenerate them whenever you want to get the latest updates/improvements.
> 
> **TLDR: we’ve removed the metadata files from our version downloads, but will work to provide a script to get the same (static) files and information via our API.**

**Comment by DanielMoberly** [comment]
This script solution is an improvement, but its not particularly viable either. 

Case in point: the Drupal integration of the Font Awesome module relies on the metadata to provide searchable icon insertion systems in Drupal. Users currently do not have to have technical knowledge to insert icons - they can use the autocomplete form to search for and insert icons they want. This is based on the search terms and the categories information provided in the metadata. Currently this information is not available anywhere for v6 (not in the API either)

The module is designed such that non-technical users (those who are often responsible for this kind of work) can simply turn on the module and drop the Font Awesome module in the libraries folder and be ready to go - the module pulls the metadata from the library itself, ensuring that it has the most up-to-date data. When users update their version of Font Awesome, the autocomplete & search functionality automatically matches, as its based on the provided metadata from the user. This also allows Free & Pro versions to work seamlessly as the available icons & terms are dependent on what metadata is available (Pro icons won't show if you only have the Free version).

What you're now asking is that a custom script be run to pull metadata. There are two ways to go about trying to solve this (from my perspective) and both of them cause issues for the users:
1) The Font Awesome module provides the metadata file to the users - this is not a viable solution as we don't know whether to provide the Pro or the Free metadata to the user. Additionally, the file becomes immediately outdated as its limited by the latest release of the module.
2) Ask the users to pull the metadata themselves with the "script". This puts a huge additional burden on the 66 thousand users who are used to simply dropping the new Font Awesome version into the libraries folder for an update. They now have to figure out how to integrate with the API and manually generate themselves a metadata file - a huge portion of this user group will not have the technical knowledge to be able to perform this extra step.

The correct solution is either to 
1) Provide the metadata files as has always been done (including in the beta leading up to the v6 release) or 
2) Provide all of the metadata you previously provided through the API; this would require adding the Search terms and Categories into the API.

If you want us to use the API as a replacement (as you seem to be expressing) - then it needs to be made feature-complete in comparison to what was provided in the metadata. If you added the search terms and categories to the API, I would have very little issue in using that instead. Your problem stems from the fact that you don't currently have a viable solution in the API, but you've gone ahead and dropped the metadata files anyway. What are we supposed to do in the interim?

Frankly I find it pretty ridiculous that the metadata files were dropped for the final release without a viable replacement solution for those that use it. Yes, I understand that it makes your internal processes easier, but it seems that little consideration was given to the processes of those who use Font Awesome  - all of the beta versions included metadata and those of us prepping for Font Awesome 6 made all of our preparations based on those beta releases.

**Comment by jwr** [comment]
@tagliala Just to explain — I'm not sure if you realize that for some of us the `icons.json` file is not really "metadata". It's Font Awesome itself, the library, it's the only way we access, process and use the icons.

From my point of view, for the use case I described above in this thread, the releases could consist of the `icons.json` file and nothing else. That's what I'm paying for and that's where the value is.

If this information can be pulled from your API through a script, why not just do this as part of the build process and include the prebuilt `icons.json` in releases? It will save everyone effort, result in fewer dependencies and less brittle systems, and reduce the load on your API.

**Comment by DanielMoberly** [comment]
The only real argument I'm seeing here for removing it is that it simplifies internal processes at Font Awesome. While that's not nothing, its a tough sell for the rest of us.

**Comment by sensibleworld** [comment]
Just a note to say that we're reading (and listening to) all of the feedback and uses cases above. Stay tuned.

**Comment by juliettef** [comment]
Since you're open to feedback, I suggest you open source the metadata for both the free and pro versions of FA 5 and 6. The metadata can only be used with Font Awesome and makes it easier to use the fonts. 

For example I use the icons.yml files in an open source project which helps people use FA: [IconFontCppHeaders](https://github.com/juliettef/IconFontCppHeaders/) generates codepoints that are included in [many C, C++ and C# projects]( https://github.com/juliettef/IconFontCppHeaders#projects-using-the-font-icon-header-files), including as part of the popular UI library [Dear ImGui](https://github.com/ocornut/imgui).

**Comment by dougbinks** [comment]
The argument above by @sensibleworld appears to be "we don't want to have to create a new FA release to fix metadata, so we now provide that by a separate channel".

This argument fails for the developers raising this issue because the metadata does not exist at all and the separate channel is not a viable solution. So instead of slightly wrong metadata we have completely missing metadata and unusable FA icons.

So to summarise: 

- For many this isn't an issue making it harder for us - this is an issue making it impossible for us to use FA at all. 
- In other words for many use cases slightly wrong meta data would be preferable to no metadata.

**Comment by jsidea** [comment]
The metadata folder is back, but the search terms are missing.
Is there a chance that the search terms will come back?

I hope the FA team will work "_hard to make upgrading to Version 6 as smooth and easy as possible_" :)
But perhaps dealing with the search terms as "open source" is no longer possible.

**Comment by robmadole** [comment]
Hello again! Yes, we've just released 6.1.0 and included in the packages and downloads is the metadata directory.

The search terms will not be returning. We talked about this internally quite a bit and ultimately made the business decision to keep them out. 

Some quick notes on this:

- The search terms have been removed but left as an empty array. This makes sure that any implementation that previously tried to use these won't break with a missing property in an object.
- We've included a new aliases property that contain backward-compatible names and unicode values. This makes lookups possible for names or unicodes that changed from version 5 to 6.
If you have any thoughts or questions let us know.

**Comment by AgentSmith0** [comment]
Please also bring back the search terms...

**Comment by DanielMoberly** [comment]
"This makes sure that any implementation that previously tried to use these won't break with a missing property in an object." 

Correct, they won't "break", they just won't have any functionality at all.

**Comment by jwr** [comment]
Thank you for bringing the `icons.json` file back (I'm not sure why this is called "metadata" when it contains actual icon data; in my application this is the only representation of FontAwesome that is used). Now I can finally migrate to version 6.

Migration proved to be painful because of all the icon renamings, but that is a discussion for another issue (feature request).

**Comment by spmsh** [comment]
Why removing the search terms? We use search terms and we translate them in several languages to allow our multilingual users to search in their own languages (mainly Chinese, Japanese, Korean, Thai) in the icon picker we developed for that purpose... I thought the statement you made was to bring back the metadata folder and files in the **same way** it was prior to V6... please bring back the search terms. thank you.

**Comment by seanreiser** [comment]
I’m one of the many Drupalists who routinely include the Font Awesome Module in their builds.  I’ve been following this issue for a while, and am disappointed with your resolution.  I have been a FA license holder since the first kickstarter and have a current FA6 license.  Since a client might not want me to do support when after I've built their site,  I have my clients purchase a pro license from you to ensure they are in compliance and to encourage further development of Font Awesome.  I know it’s not a lot of licenses but since you released FA4, I’ve sold a couple of dozen on your behalf.

I am now in the position of asking clients of ours to choose between the new icons and styles in FA6 and the integrated search experience that I was able to provide with FA5, which you can imagine is uncomfortable.  A number of web apps I’ve built sit on private networks so even if I wanted to spend the time and build something to leverage your new search api, it won’t work for those clients.  I have to be honest, part of me is concerned that in a future version you’ll make pro icons available only over your CDN shutting out clients on private networks further.

All that said, and as you mention making the best business decision, I need to rethink the benefits vs the risks of including FA in future projects.

**Comment by Molkobain** [comment]
> I’m one of the many Drupalists who routinely include the Font Awesome Module in their builds. I’ve been following this issue for a while, and am disappointed with your resolution. I have been a FA license holder since the first kickstarter and have a current FA6 license. Since a client might not want me to do support when after I've built their site, I have my clients purchase a pro license from you to ensure they are in compliance and to encourage further development of Font Awesome. I know it’s not a lot of licenses but since you released FA4, I’ve sold a couple of dozen on your behalf.
> 
> I am now in the position of asking clients of ours to choose between the new icons and styles in FA6 and the integrated search experience that I was able to provide with FA5, which you can imagine is uncomfortable. A number of web apps I’ve built sit on private networks so even if I wanted to spend the time and build something to leverage your new search api, it won’t work for those clients. I have to be honest, part of me is concerned that in a future version you’ll make pro icons available only over your CDN shutting out clients on private networks further.
> 
> All that said, and as you mention making the best business decision, I need to rethink the benefits vs the risks of including FA in future projects.

We are in the exact same situation 😕

**Comment by michaelspiss** [comment]
Font awesome's flexibility was the main reason I kept recommending it to my customers. Lots of icons, and if they required any customization, I was able to do it.

Think of an app that teaches to read or to understand texts in different languages. It occasionally sides words with an icon, so it gets easier to remember them. This simple use case would require hundreds if not thousands of calls to your API per use.

There are more use cases than a simple icon picker.

I'd totally understand search terms being part of the pro package only. They require constant work, so why not charge for it? No problem. But this online-only solution breaks so many business use cases, for reasons (paying) customers may find hard to understand.

**Comment by hudri** [comment]
As a consequence of Schrems II ruling any data transfer to the US without prior consent is violating the GDPR. We already have court verdicts in Belgium, Austria and Germany that the sheer usage of US cloud services and CDNs without prior consent is a violation of the GDPR, a similar verdict is expected in France soon (e.g. [Google Fonts verdict by Munich court](https://support.cookiebot.com/hc/en-us/articles/4408570605980-Require-consent-before-loading-Google-Fonts) (in english by Cookiebot) and the corresponding [official verdict](https://rewis.io/urteile/urteil/lhm-20-01-2022-3-o-1749320/) in german language)

So if your business decision is "use our CDN and our API", then your business decision also means "we no longer support the european market at all". Anything else than a fully self-hosted FontAwesome - including all icons and all the metadata - will not be an option for europeans.

**Comment by sensibleworld** [comment]
First of all, thanks to everyone for the feedback. We're sorry for the frustration caused by removing the files initially, and for the frustration still being experienced by those of you who are dependent on the search terms.

For folks who this is directly impacting, could you reach out (feedback@fontawesome.com) with details on your specific implementations? We'd love to learn more and help to figure out a solution that works.

**Comment by spmsh** [comment]
> First of all, thanks to everyone for the feedback. We're sorry for the frustration caused by removing the files initially, and for the frustration still being experienced by those of you who are dependent on the search terms.
> 
> For folks who this is directly impacting, could you reach out ([feedback@fontawesome.com](mailto:feedback@fontawesome.com)) with details on your specific implementations? We'd love to learn more and help to figure out a solution that works.

Sending an email to feedback as below:

We have developed our own version of the icon picker that comes in a multilingual fashion, and we are using **and translating** search terms so users can use the icon picker and search icons using their own languages. With the missing search terms in V6, the icon picker is simply unusable for us, it's a lot of work we have to throw away and not really willing to redo anything from scratch.

Even if we do redevelop something, using your API is not an option for us as our software is not cloud based and very seldom has internet connectivity. We mostly deploy our software on-premise in closed network so API would not be an option for us.

So can you please **PLEASE** bring back the search terms? Many thanks.

**Comment by meiyasan** [comment]
@sensibleworld 

Common, we basically all have the same pattern as @spmsh..
What a waste of time here.. Couldn't you have done anything better from Feb 9th to today.. such as a GraphQL pre-request to provide a partial metadata file.. Just make partial dump including search box in your GraphQL database. Is that so hard ?
Who cares about raw SVG information in metadata file..

Really disappointed, I will for sure avoid using and paying for such toolset in the future..
This is at the opposite of backward compatibility principles, without any real deprecation and timeline: unprofessional

**Comment by vadimpopa** [comment]
Big disappointment. Now we have to tell to our clients too, that they cannot search anymore by terms. In enterprise world this is a big breaking change.

**Comment by apierri** [comment]
Hello. Could you provide to your pro users (myself included) the reason behind the "business decision" to exclude search terms, even for Pro members? It would be useful information to consider next time we're deciding to continue (or not) backing this project.

**Comment by AgentSmith0** [comment]
I also want the search terms back, they were very useful for out intern website builder, to quickly find icons. Very sad, you removed them.

**Comment by vadimpopa** [comment]
We've decided not to upgrade to v6 cause of this. We will upgrade only if/when search terms will be added back.

**Comment by silasgcf** [comment]
The search terms are very useful for our application. Please consider coming back with this resource.

**Comment by denny99** [comment]
We tried to implement our own icon picker and stumbled upon this issue too. 

The graphql API works and we could use that for the search functionality, but this would require our application to always have a connection to the outside world. 
Most of our customers reject such a situation due to security reasons, so our application will always run in internal networks and therefore cannot access the graphql api at all.

**Comment by meiyasan** [comment]
Hi @denny99
By any chance, do you have kind of a practical example to run with FA graph ql ?
I saw that FA team provides some kind of metadata.json, I would like to reproduce. I wonder if we can get some search term out of it.

**Comment by denny99** [comment]
@xKZL I can construct you one on Monday with stackblitz if you want to

**Comment by denny99** [comment]
@xKZL 
https://stackblitz.com/edit/primeng-autocomplete-demo-ugd3lm?file=src%2Fapp%2Fapp.component.html,src%2Fapp%2Fapp.component.ts,package.json,src%2Fapp%2Fapp.module.ts,angular.json

Here you go. I abused the primeng autocomplete demo for this :D it's not exactly what we are doing (our picker can also select style, animations etc.) But for the basics this should be it.

You can type into the input and it searches with graph ql for icons (only free, as the others won't be displayed)

<img width="329" alt="image" src="https://user-images.githubusercontent.com/1826849/200273332-645b78b6-ba60-40ed-b6f3-7e7cd16e47ce.png">

**Comment by sensibleworld** [comment]
Hey folks! Wanted to give a quick update that, as of our [6.3.0 release](https://fontawesome.com/docs/changelog/), we've added back the search terms to our metadata files, where they'll stay. Apologies for the poor communication and frustration we caused around this matter.

**Comment by Molkobain** [comment]
Thank you! 🙌❤

**Comment by meiyasan** [comment]
Yayy thank you so much !
(lmfao, this thread is 1y old but one day.)
